newrelic/resource_newrelic_synthetics_monitor.go

```diff
@@ -95,158 +94,411 @@ func resourceNewRelicSyntheticsMonitor() *schema.Resource {
+	}
+
+	if v, ok := d.GetOk("verify_ssl"); ok {
+		simpleBrowserMonitorUpdateInput.AdvancedOptions.UseTlsValidation = v.(bool)
+	}
+
+	if v, ok := d.GetOk("script_language"); ok {
+		simpleBrowserMonitorUpdateInput.Runtime.ScriptLanguage = v.(string)
+	}
+
+	if v, ok := d.GetOk("runtime_type"); ok {
+		simpleBrowserMonitorUpdateInput.Runtime.RuntimeType = v.(string)
+	}
+
+	if v, ok := d.GetOk("runtime_type_version"); ok {
+		simpleBrowserMonitorUpdateInput.Runtime.RuntimeTypeVersion = synthetics.SemVer(v.(string))
+	}
+
+	return simpleBrowserMonitorUpdateInput
+}
+
+//func to build input to update simple monitor
+func buildSyntheticsSimpleMonitorUpdateStruct(d *schema.ResourceData) synthetics.SyntheticsUpdateSimpleMonitorInput {
+	simpleMonitorUpdateInput := synthetics.SyntheticsUpdateSimpleMonitorInput{}
+
+	inputBase := expandSyntheticsMonitorBase(d)
+
+	simpleMonitorUpdateInput.AdvancedOptions.CustomHeaders = inputBase.CustomHeaders
+	simpleMonitorUpdateInput.Name = inputBase.Name
+	simpleMonitorUpdateInput.Period = inputBase.Period
+	simpleMonitorUpdateInput.Status = inputBase.Status
+	simpleMonitorUpdateInput.Tags = inputBase.Tags
+	simpleMonitorUpdateInput.Uri = inputBase.URI
+
+	if v, ok := d.GetOk("locations_public"); ok {
+		simpleMonitorUpdateInput.Locations.Public = expandSyntheticsSimplePublicLocations(v.(*schema.Set).List())
+	}
+
+	if v, ok := d.GetOk("locations_private"); ok {
+		simpleMonitorUpdateInput.Locations.Private = expandSyntheticsSimplePrivateLocations(v.(*schema.Set).List())
+	}
+
+	if v, ok := d.GetOk("treat_redirect_as_failure"); ok {
+		simpleMonitorUpdateInput.AdvancedOptions.RedirectIsFailure = v.(bool)
+	}
+
+	if v, ok := d.GetOk("validation_string"); ok {
+		simpleMonitorUpdateInput.AdvancedOptions.ResponseValidationText = v.(string)
+	}
+
+	if v, ok := d.GetOk("bypass_head_request"); ok {
+		simpleMonitorUpdateInput.AdvancedOptions.ShouldBypassHeadRequest = v.(bool)
+	}
+
+	if v, ok := d.GetOk("verify_ssl"); ok {
+		simpleMonitorUpdateInput.AdvancedOptions.UseTlsValidation = v.(bool)
+	}
+
+	return simpleMonitorUpdateInput
+
+}
+
 func resourceNewRelicSyntheticsMonitorUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
 	client := meta.(*ProviderConfig).NewClient
+
 	log.Printf("[INFO] Updating New Relic Synthetics monitor %s", d.Id())
 
-	_, err := client.Synthetics.UpdateMonitorWithContext(ctx, *buildSyntheticsUpdateMonitorArgs(d))
-	if err != nil {
-		return diag.FromErr(err)
+	var diags diag.Diagnostics
+
+	monitorType, ok := d.GetOk("type")
+	if !ok {
+		log.Printf("Not Monitor type specified")
 	}
 
-	return resourceNewRelicSyntheticsMonitorRead(ctx, d, meta)
+	guid := synthetics.EntityGUID(d.Id())
+
+	switch monitorType.(string) {
+
+	case "SIMPLE":
+		simpleMonitorUpdateInput := buildSyntheticsSimpleMonitorUpdateStruct(d)
+
+		resp, err := client.Synthetics.SyntheticsUpdateSimpleMonitorWithContext(ctx, guid, simpleMonitorUpdateInput)
+
+		if err != nil {
```

Comment:
Same here, the error handling code is same in both cases. Could we move it outside of the switch case?